index.d.ts

```diff
@@ -0,0 +1,469 @@
+// Type definitions for realm-js 1.3.1
+// Project: https://github.com/realm/realm-js
+// Definitions by: Akim <https://github.com/Akim95> & Maximilian Alexander <https://github.com/mbalex99> 
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+declare namespace Realm {
+    /**
+     * PropertyType
+     * @see { @link https://realm.io/docs/javascript/latest/api/Realm.html#~PropertyType }
+     */
+    type PropertyType = string | 'bool' | 'int' | 'float' | 'double' | 'string' | 'data' | 'date' | 'list';
```

Comment:
The docs specify only 

Type: "bool" or "int" or "float" or "double" or "string" or "date" or "data" or "list" or "<ObjectType>"

"<ObjectType>" is a string for another name of an object. Thus, that's why this TypeScript string literal has an accepted `string` native type.